modules/core/include/opencv2/core/hal/intrin_avx.hpp

```diff
@@ -650,13 +656,83 @@ OPENCV_HAL_IMPL_AVX_BIN_OP(-, v_float64x4, _mm256_sub_pd)
 OPENCV_HAL_IMPL_AVX_BIN_OP(*, v_float64x4, _mm256_mul_pd)
 OPENCV_HAL_IMPL_AVX_BIN_OP(/, v_float64x4, _mm256_div_pd)
 
+// saturating multiply 8-bit
+inline v_uint8x32 operator * (const v_uint8x32& a, const v_uint8x32& b)
+{
+    __m256i a0 = _mm256_cvtepu8_epi16(_v256_extract_low(a.val));
+    __m256i a1 = _mm256_cvtepu8_epi16(_v256_extract_high(a.val));
+    __m256i b0 = _mm256_cvtepu8_epi16(_v256_extract_low(b.val));
+    __m256i b1 = _mm256_cvtepu8_epi16(_v256_extract_high(b.val));
+    __m256i p0 = _mm256_mullo_epi16(a0, b0);
+    __m256i p1 = _mm256_mullo_epi16(a1, b1);
+    return v_uint8x32(_v256_shuffle_odd_64(_mm256_packus_epi16(p0, p1)));
+}
+inline v_int8x32 operator * (const v_int8x32& a, const v_int8x32& b)
+{
+    __m256i a0 = _mm256_cvtepi8_epi16(_v256_extract_low(a.val));
+    __m256i a1 = _mm256_cvtepi8_epi16(_v256_extract_high(a.val));
+    __m256i b0 = _mm256_cvtepi8_epi16(_v256_extract_low(b.val));
+    __m256i b1 = _mm256_cvtepi8_epi16(_v256_extract_high(b.val));
+    __m256i p0 = _mm256_mullo_epi16(a0, b0);
+    __m256i p1 = _mm256_mullo_epi16(a1, b1);
+    return v_int8x32(_v256_shuffle_odd_64(_mm256_packs_epi16(p0, p1)));
+}
+// saturating multiply 16-bit
+inline v_uint16x16 operator * (const v_uint16x16& a, const v_uint16x16& b)
```

Comment:
in general, all the saturation `*` operations can be implemented as v_mul_expand() + v_pack().

Reply:
Exactly except avx2::uint16 & int64 due performance lose, but I had to implement forward declarations for theses intrinsics in order to re-implement  saturation * operations .